Thistle/galley-sdk
# (ENG-1674) Refactor query for recipe_connection_query

## Description
- Refactors `recipe_connection_query` and `get_formatted_recipes_data` that powers all recipe data thistle-web receives to reduce inefficiencies caused by querying nested `recipeTreeComponents`, and `quantityUnitValues`
- Query pulls in new fields `ingredientsWithUsages`, `usages.ancestorRecipes`, and `unit.UnitValues`

**Note:** Mock tests/data related to `recipeTreeComponents` is still currently being used by the ops menu data query, which will be refactored in a subsequent PR. For the changes made in this PR, any real/mock data related to `recipeTreeComponents` should be expunged. 


## Test Plan
1. Checkout main
2. Open python shell and run:
```python
from galley.formatted_queries import *; from pprint import pprint

formatted_menus = get_formatted_menu_data(['2023-04-10', '2023-04-13', '2023-04-17', '2023-04-20', '2023-04-24', '2023-04-27'], 'Vacaville')
formatted_recipes = get_formatted_recipes_data([mi['recipeId'] for m in formatted_menus for mi in m['menuItems']], 'Vacaville')

pprint(formatted_menus)
pprint(formatted_recipes)
```
3. Copy and paste output in a [diff checker](https://www.diffchecker.com/)
4. Checkout this branch
5. Repeat Step 2 and Step 3 to compare the diff

**Note:** If you're seeing slight mismatches between a few ingredient list diffs, the issue is most likely due to slight float precision differences caused from using `quantityUnitValues` (previously) and `unitValues` (currently) for ingredient calculations. You can verify the ingredient list for a recipe in Galley with their sort by usages feature, but keep in mind that Galley often rounds their ingredient usage values shown in the UI, and the difference should be a negligible amount. For context, our ingredient lists are primary sorted by usage in descending order, and then secondary sorted by external (common) name in ascending order, as required by the FDA. 

**Note:** Our "Ingredients" object returned by `get_formatted_recipes_data` does _not_ include standalone ingredients, so if using the Galley UI to compare "Ingredients" values at the _sellable_ recipe level, remember to factor out standalone usages.

- [ ] Verify `get_formatted_menu_data` in v0.45.0 returns the same shape and data values as it does in v0.44.0
- [ ] Verify `get_formatted_recipes_data` in v0.45.0 returns the same shape and data values as it does in v0.44.0


## Versioning
0.44.0 → 0.45.0
